src/components/elements/AutoCompleteMenu.js

```diff
@@ -0,0 +1,35 @@
+import React from 'react'
+import AutoComplete from 'material-ui/AutoComplete'
+import * as formActions from '../../actions/formActions'
+import * as filterActions from '../../actions/filterActions'
+
+const dataSourceConfig = {
+    text: 'name',
+    value: 'id'
+}
+
+const AutoCompleteMenu = ({dispatch, history, label, menuOptions, filter, filterSearchField, menuFilterSearchForm}) => {
+    let filterSearchForm = menuFilterSearchForm
+    return filter? (
+        <div>
+            <AutoComplete
+                floatingLabelText={label}
+                filter={AutoComplete.caseInsensitiveFilter}
+                dataSource={menuOptions}
+                searchText={filterSearchField}
+                onUpdateInput={(searchText) => dispatch(formActions.updateInput(filterSearchForm.name, filterSearchForm.field, searchText))}
+                onNewRequest={
```

Comment:
There's still a bug in onNewRequest, because the object passed into the function can be one of two things: when someone clicks on the list, it gets the eventType object selected (which works), but when someone presses enter, it gets just the text (example: 'Note'), and dispatch fails.

I suggest pulling out the onNewRequest function into a defined const with a signature like

const onNewRequest = (filterTypes) => (inputResult) => {

Which makes the argument to AutoComplete:

onNewRequest={onNewRequest(filterTypes)}

Then in onNewRequest you can check whether inputResult is an object or a string, and if it's a string you can use the string to lookup the eventType by name in filterTypes.